src/NSCallBacks.m

```diff
@@ -237,3 +237,37 @@
   /* Is this useful? */
   return [NSString stringWithFormat: @"%d(%#"PRIxPTR")", *p, (NSUInteger)p];
 }
+
+/** For owned Objective-C objects **/
+
+NSUInteger
+_NS_owned_id_hash(void *table, id <NSObject> o)
+{
+  /* P may be aligned, so we need to compensate. */
+  return [o hash];
+}
+
+BOOL
+_NS_owned_id_is_equal(void *table, id <NSObject> p)
+{
+  return [o isEqual: p];;
```

Comment:
Object `o` does not exist in this function. Also you have two semicolons.

Reply:
True, thanks.